parts/linux/cloud-init/artifacts/bind-mount.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+
+# Bind mount kubelet to ephemeral storage on startup, as necessary.
+#
+# This fixes an issue with kubelet's ability to detect allocatable
+# capacity for Node ephemeral-storage. On Azure, ephemeral-storage
+# should correspond to the temp disk if a VM has one. This script makes
+# that true by bind mounting the temp disk to /var/lib/kubelet, so
+# kubelet thinks it's located on the temp disk (/dev/sdb). This results
+# in correct calculation of ephemeral-storage capacity.
+
+{{if eq GetKubeletDiskType "Temporary"}}
```

Comment:
shouldn't we skip the whole script if it's not Temporary disk?

Reply:
We do in cloud init, but we still need the value here. This could theoretically be expanded to account for nvme which has come up in the last.

Reply:
I see what you’re saying, since this will be in the vhd. I think we only need it in cloud init, really.

Reply:
i'm going to leave this as is for now, but we might actually be able to take advantage of this to replace https://github.com/Azure/AgentBaker/blob/c6f9bd70f13173ac78ca56ed79057b8b796e10c1/parts/linux/cloud-init/artifacts/kubelet.service#L15

Reply:
I don't like this mixed-now-and-future approach as it confusing. Let's do it this way
1. rename this file and service to include "tempdisk" in their name
2. make this script specific for temp disk case
3. remove HasKubeletDiskType and GetKubeletDiskType, and add a new one, say IsTempDiskForKubelet (whatever you like better) and use it in all places when necessary

My intention here is to be specific and reduce confusion. If we need to support other cases, let's add it later.

Can we do this as part of the refactoring that we talked about earlier?

Reply:
Sure, makes sense to me